src/python/pants/backend/scala/compile/scalac.py

```diff
@@ -146,31 +128,27 @@ async def compile_scala_source(
                 ),
             ),
         ),
-        Get(
-            MaterializedClasspath,
-            MaterializedClasspathRequest(
-                prefix="__thirdpartycp",
-                lockfiles=(lockfile,),
-            ),
-        ),
         Get(
             Digest,
-            MergeDigests(classfiles.digest for classfiles in transitive_dependency_classfiles),
+            # Flatten the entire transitive classpath.
```

Comment:
Should we be worried about classpath conflicts at this point? E.g., same class from different sources.

Reply:
They are always possible. But because we have wrapping jars in this case, the classfiles will not collide the way you might be thinking? The jar filenames would have to collide.

Reply:
Yeah was unclear to me whether they were jars at this point or exploded files. So as per your point that it is jar files, my comment is moot.

Reply:
Yea, the word "classfiles" is still scattered around in variables names a bit after #13508. But `ClasspathEntry` now always contains a series of jar files, and never loose classfile directories.